k2/python/k2/fsa.py

```diff
@@ -26,6 +26,57 @@ class Fsa(object):
     a vector of FSAs it is a tuple with three
     elements ``(num_fsas, None, None)``.  (Caution: it's possible
     for a vector of FSAs to have zero or one elements).
+
+    An instance of FSA has the following attributes:
+
+    - ``arcs``: You will NOT use it directly in Python. It is an instance
+                of ``_k2.RaggedArc`` with only one method ``values()`` which
+                returns a 2-D `torch.Tensor`` of dtype ``torch.int32`` with 4
+                columns. Its number of rows indicates the number of arcs in the
+                FSA. The first column represents the source states, second
+                column the destination states, third column the labels and the
+                fourth column is the score. Note that the score is actually
+                a float number but it is **reinterpreted** as an integer.
+
+    - ``score``: A 1-D ``torch.Tensor`` of dtype ``torch.float32``. It has
+                 as many entries as the number of arcs representing the score
+                 of every arc.
+
+    - ``labels``: A 1-D ``torch.Tensor`` of dtype ``torch.int32``. It has as
+                  many entries as the number of arcs representing the label of
+                  every arc.
+
+    - ``arc_labels_sorted``: A boolean. True if arcs of every state are sorted
+                             by labels in increasing order. False means arcs
+                             are not sorted by ``labels``.
+
+    It MAY have the following attributes:
+
+    - ``symbols``: An instance of ``k2.SymbolTable``. It maps an entry in
+                   ``labels`` to an integer and vice versa. It is used for
+                   visualization only.
+
+    - ``aux_labels`: A 1-D ``torch.Tensor`` of dtype ``torch.int32``. It has the
```

Comment:
not sure if we should add a TODO here to say, it maybe a `RaggedTensor` in the future.

Reply:
Added.